src/Sylius/Component/Core/Promotion/Filter/ProductFilter.php

```diff
@@ -21,7 +21,7 @@
      */
     public function filter(array $items, array $configuration)
     {
-        if (!isset($configuration['filters']['products_filter'])) {
+        if (!isset($configuration['filters']['products_filter']) || empty($configuration['filters']['products_filter']['products'])) {
```

Comment:
`!isset($configuration['filters']['products_filter']['products'])` would be enough